### ISSUE TITLE
Should no-irregular-whitespace skip template strings by default?

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#10160]
**What rule do you want to change?**

no-irregular-whitespace

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior

**Please provide some example code that this change will affect:**

```js
/* eslint no-irregular-whitespace: error */

` <-- non-breaking space`;
```

**What does the rule currently do for this code?**

3:2 - Irregular whitespace not allowed. (no-irregular-whitespace)

**What will the rule do after it's changed?**

No error (template strings will be skipped by default, just like other strings)

-----------

This is currently working as intended since we have `skipTemplates: true` option available. The question here is, should we change the default behavior with the 5.0 release so that template strings are skipped by default? Seems like this would be the more natural behavior given that `skipStrings` is also true by default. My guess is we added `skipTemplates` in a semver-minor release and just never went back to change the default behavior around templates.

The change would be to have `skipTemplates` default to `true`.

Inspired by #10159.

**Comment by pfgithub** [comment]
Shouldn't this use the feature request format instead of bug report format?

**Comment by platinumazure** [comment]
Probably, but I was too tired to change it. I'll edit it later. :smile:

**Comment by platinumazure** [comment]
@pfgithub Fixed to use rule change request template, thanks.

**Comment by not-an-aardvark** [comment]
What's the reasoning for why `skipStrings` is currently the default? To me, it seems like the rule shouldn't allow whitespace in any type of string by default.
